subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationWritingIntegTest.groovy

```diff
@@ -595,15 +596,14 @@ class DependencyVerificationWritingIntegTest extends AbstractDependencyVerificat
       <component group="org" name="foo" version="1.0">
          <artifact name="foo-1.0.jar">
             <md5 value="abc"/>
-            <sha1 value="1234"/>
-         </artifact>
-         <artifact name="foo-1.0.jar">
-            <sha1 value="16e066e005a935ac60f06216115436ab97c5da02"/>
-            <sha512 value="734fce768f0e1a3aec423cb4804e5cdf343fd317418a5da1adc825256805c5cad9026a3e927ae43ecc12d378ce8f45cc3e16ade9114c9a147fda3958d357a85b"/>
+            <sha1 value="1234" origin="Generated by Gradle">
+               <also-trust value="16e066e005a935ac60f06216115436ab97c5da02"/>
```

Comment:
👍 Nice and great for the `origin` attribute as well!

But that makes me wonder if there should not be a `origin` attribute for the `also-trust` node. That would allow document each entry with regards to its origin.

Reply:
I didn't do this mostly for performance, in order to avoid a too expensive datastructure. I think if the user wants to document this they can use the parent `origin`.

Reply:
Fair point, all good then!